src/MySqlConnector/Protocol/Payloads/ColumnCountPayload.cs

```diff
@@ -0,0 +1,28 @@
+using MySqlConnector.Protocol.Serialization;
+
+namespace MySqlConnector.Protocol.Payloads;
+
+/// <summary>
+///     Helper class to parse Column count packet.
+///     https://mariadb.com/kb/en/result-set-packets/#column-count-packet
+///     Packet contains a columnCount, and - if capability MARIADB_CLIENT_CACHE_METADATA is set - a flag to indicate if metadata follows
+/// </summary>
+internal sealed class ColumnCountPayload
```

Comment:
Note to self: Most other `Payload` objects are `class`es, but would it be worthwhile making this (and others?) a `readonly struct`?

Reply:
https://github.com/mysql-net/MySqlConnector/commit/3e746d8af2fc7e17bee26dbbc7503f7b3570eb1f